protocol/messenger_communities.go

```diff
@@ -4380,14 +4380,19 @@ func (m *Messenger) GetCommunityMembersForWalletAddresses(communityID types.HexB
 
 	membersForAddresses := map[string]*Contact{}
 
-	for memberID, member := range community.Members() {
-		for _, revealedAddress := range member.RevealedAccounts {
-			if slices.Contains(revealedAddress.ChainIds, chainID) {
-				contact, ok := m.allContacts.Load(memberID)
+	for _, memberPubKey := range community.GetMemberPubkeys() {
+		memberPubKeyStr := common.PubkeyToHex(memberPubKey)
+		revealedAccounts, err := m.communitiesManager.GetRevealedAddresses(communityID, memberPubKeyStr)
+		if err != nil {
+			return nil, err
+		}
+		for _, revealedAccount := range revealedAccounts {
+			if slices.Contains(revealedAccount.ChainIds, chainID) {
```

Comment:
I'd appreciate short-circuit continue, but that's not important. Just less spaghetti 🙂 
```go
if !slices.Contains(revealedAccount.ChainIds, chainID) {
    continue
}
...
```

Reply:
fixed!